### PR TITLE
Fix cron: add cron task only once and start cron service

### PR DESCRIPTION
**Pull request by YannickR26**
**Fix:**
- the cron service was not started at startup
- allows to leave the environment variable DOLI_CRON at 1

### CODE REVIEW

**Comment by tuxgasy** [comment]
If you change DOLI_CRON_KEY or DOLI_CRON_USER, the cron task will not be updated.

**Comment by YannickR26** [comment]
Yes, its true, we can update the cron task at each run if you prefer ?

**Comment by tuxgasy** [comment]
Currently, the cron file is already regenerated at each run.

**Comment by YannickR26** [comment]
ok, I updated the `docker-run.sh` file to check the environment variables and create the cron job at each run.

File: docker-run.sh [hunk 1]
```diff
@@ -231,10 +231,26 @@ run
...
+  fi
+  if [[ -z ${DOLI_CRON_KEY} ]]; then
+    echo "'DOLI_CRON_KEY' is undefined or empty"
+    echo "Unable to create cron task !"
+    exit 1
+  fi
+  # Remove old file
+  if [[ -f ${DOLI_CRON_FILE} ]]; then
+    rm -rf ${DOLI_CRON_FILE}
+  fi
```
Comment (tuxgasy) [inline docker-run.sh:248]: It's not necessary because the content will be overwritten by the first echo.

File: docker-run.sh [hunk 2]
```diff
@@ -231,10 +231,26 @@ run
 set -e
 
 if [[ ${DOLI_CRON} -eq 1 ]]; then
-    echo "PATH=\$PATH:/usr/local/bin" > /etc/cron.d/dolibarr
-    echo "*/5 * * * * www-data /var/www/scripts/cron/cron_run_jobs.php ${DOLI_CRON_KEY} ${DOLI_CRON_USER} > /var/www/documents/cron_run_jobs.php.log 2>&1" >> /etc/cron.d/dolibarr
-    cron -f
-    exit 0
+  DOLI_CRON_FILE=/etc/cron.d/dolibarr
```
Comment (tuxgasy) [inline docker-run.sh:234]: I'm not sure if it's really necessary to create a variable for that.

File: docker-run.sh [hunk 3]
```diff
@@ -198,7 +198,7 @@ function run()
   initDolibarr
   echo "Current Version is : ${DOLI_VERSION}"
 
-  if [[ ${DOLI_INSTALL_AUTO} -eq 1 && ${DOLI_CRON} -ne 1 && ! -f /var/www/documents/install.lock && ${DOLI_DB_TYPE} != "pgsql" ]]; then
+  if [[ ${DOLI_INSTALL_AUTO} -eq 1 && ! -f /var/www/documents/install.lock && ${DOLI_DB_TYPE} != "pgsql" ]]; then
```
Comment (tuxgasy) [inline docker-run.sh:201]: The database update should be done only from the main container.

File: docker-run.sh (hunk 1 above)
Comment (YannickR26) [reply docker-run.sh:248]: Yes, it's true! It was just to be clearer

File: docker-run.sh (hunk 2 above)
Comment (YannickR26) [reply docker-run.sh:234]: is to be cleaner !

File: docker-run.sh (hunk 1 above)
Comment (Vaadasch) [reply docker-run.sh:248]: You don't have to delete it only if it is there, you don't have to test, you can do `rm -rf ${DOLI_CRON_FILE}` without the if.

**Comment by tuxgasy** [comment]
Please open another PR if needed.

**Pull request closed without merging**